addons/modules/functions/fnc_moduleTransferOwnership.sqf

```diff
@@ -0,0 +1,106 @@
+#include "script_component.hpp"
+/*
+ * Author: Ampersand
+ * Zeus module function to transfer ownership of objects and groups.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call zen_modules_fnc_moduleTransferOwnership
+ *
+ * Public: No
+ */
+
+params ["_logic"];
+deleteVehicle _logic;
+
+if (!isMultiplayer) exitWith {
+    [LSTRING(OnlyMultiplayer)] call EFUNC(common,showMessage);
+};
+
+private _entities = [];
+(call EFUNC(editor,getSelection)) params ["_objects", "_groups"];
+{
+    if (isNull group _x) then {
+        _entities pushBack _x;
+    } else {
+        _groups pushBackUnique group _x;
+    };
+} forEach _objects;
+_entities append _groups;
+
+if (_entities findIf {units _x findIf {isPlayer _x} > -1} != -1) exitWith {
+    [LSTRING(SelectionCannotIncludePlayers)] call EFUNC(common,showMessage);
+};
+
+private _targets = [2, clientOwner];
+
+private _targetNames = [
+    LSTRING(ModuleTransferOwnership_Server),
+    "str_a3_cfgvehicles_module_f_moduledescription_curator_f_1"
+];
+
+private _HCs = [];
+private _players = [];
+{
+    if (_x isKindOf "HeadlessClient_F") then {
+        _HCs pushBack [name _x, _x];
+    } else {
+        _players pushBack [name _x, _x];
+    };
+} forEach allPlayers;
+_HCs sort true;
+_players sort true;
+{
+    _x params ["_name", "_entity"];
+    _targetNames pushBack _name;
+    _targets pushBack _entity;
+} forEach (_HCs + _players);
+
+// Set default target to curator, server, or HC depending on current locality
+private _defaultTarget = if (local (_entities select 0)) then {
+    if (_HCs isEqualTo []) then {
+        0
+    } else {
+        2
+    };
+} else {
+    1
+};
+
+[LSTRING(ModuleTransferOwnership), [
+    [
+        "COMBO",
+        ELSTRING(common,Target),
+        [_targets, _targetNames, _defaultTarget],
+        true
+    ],
+    [
+        "TOOLBOX",
+        LSTRING(ModuleTransferOwnership_HCScripts),
```

Comment:
Tooltip, at least, please? What is a HC script?

Reply:
It does auto-select based on the current locality:
- If local, assume target will be HC and balancing will be enabled.
- If remote, assume target will be Curator and balancing will be disabled.
However I wanted to let user pick for edge cases.
a1696ce